From 9fa18765de06517697662058e0117d8ea61acce1 Mon Sep 17 00:00:00 2001
From: Nikita Yushchenko <nikita.yoush@cogentembedded.com>
Date: Fri, 26 Apr 2024 16:47:22 +0500
Subject: [PATCH 100/157] net: ethernet: renesas: rswitch: handle phy ioctls
 and ethtool ops

- pass unknown netdev ioctls to phy_mii_ioctl(), thus supporting phy
  ioctls on sockets,

- add default handlers for [sg]et_link_ksettings ethtool ops, thus
  allowing phy configuration over ethtool.

Signed-off-by: Nikita Yushchenko <nikita.yoush@cogentembedded.com>

diff --git a/drivers/net/ethernet/renesas/rswitch.c b/drivers/net/ethernet/renesas/rswitch.c
index fa3821b0f726..eeb3a14a9ed9 100644
--- a/drivers/net/ethernet/renesas/rswitch.c
+++ b/drivers/net/ethernet/renesas/rswitch.c
@@ -2467,10 +2467,11 @@ static int rswitch_do_ioctl(struct net_device *ndev, struct ifreq *req, int cmd)
 	case SIOCSHWTSTAMP:
 		return rswitch_hwstamp_set(ndev, req);
 	default:
-		break;
+		if (ndev->phydev)
+			return phy_mii_ioctl(ndev->phydev, req, cmd);
 	}
 
-	return 0;
+	return -EOPNOTSUPP;
 }
 
 static const struct net_device_ops rswitch_netdev_ops = {
@@ -2502,6 +2503,8 @@ static int rswitch_get_ts_info(struct net_device *ndev, struct ethtool_ts_info *
 
 static const struct ethtool_ops rswitch_ethtool_ops = {
 	.get_ts_info = rswitch_get_ts_info,
+	.get_link_ksettings = phy_ethtool_get_link_ksettings,
+	.set_link_ksettings = phy_ethtool_set_link_ksettings,
 };
 
 static const struct of_device_id renesas_eth_sw_of_table[] = {
-- 
2.25.1

